### DISCUSSION TITLE
Accessing Chrome Data On Mac Without Having to Enter Password

### DISCUSSION THREAD

**Discussion by nounouthereal** [moonD4rk/HackBrowserData#318] [Q&A] [answered]
## Feature Description
I would like to know if it exists a way to access chrome data/masterkey using the executable on latest MacOS versions without having to enter the system password (avoid exit code 128)

## Why is this feature needed?
Avoid interaction with the victim + Possibility to have a fully silent executable
## Checklist
--Empty--
## Screenshots/Videos
<img width="432" alt="Screenshot 2024-03-05 at 13 49 06" src="https://github.com/moonD4rk/HackBrowserData/assets/100050693/20c15402-adea-43e8-b164-0d7370dd3c19">

## Additional Context
--Empty--

**Comment by nounouthereal** [comment]
.

**Comment by moonD4rk** [comment, marked as answer]
Your proposal is very good. I have also considered this issue, but as far as I know, it is impossible for ordinary users to access the macOS system's keychain without a password. It may only be possible by using some security vulnerabilities on the macOS system to gain root privileges.

**Reply by ZH3FENG** [reply]
I think we might find answer from chrome source code. How chrome access master key in keychain？

**Reply by ZH3FENG** [reply]
I got it, chrome is allowed to access the keychain item.
(https://developer.apple.com/documentation/security/keychain_services/access_control_lists)

**Comment by moonD4rk** [comment]
> I got it, chrome is allowed to access the keychain item. (https://developer.apple.com/documentation/security/keychain_services/access_control_lists)

Yep! You can take a look at this project on [Keybase](https://github.com/keybase/go-keychain). After the process writes an item to the keychain, it does not require a password to retrieve it. Chrome also uses a similar method.
